packages/@aws-cdk/aws-lambda/lib/layer-version.ts

```diff
@@ -0,0 +1,163 @@
+  /**
+   * The description the this Lambda Layer.
+   */
+  description?: string;
+
+  /**
+   * The SPDX licence identifier or URL to the license file for this layer.
+   */
+  license?: string;
+
+  /**
+   * The name of the layer.
+   * @default a name will be generated.
+   */
+  name?: string;
+}
+
+/**
+ * A reference to a Lambda Layer version.
+ */
+export abstract class LayerVersionRef extends cdk.Construct {
+  /**
+   * Imports a Layer that has been defined externally.
+   *
+   * @param parent the parent Construct that will use the imported layer.
+   * @param id     the id of the imported layer in the construct tree.
+   * @param props  the properties of the imported layer.
+   */
+  public static import(parent: cdk.Construct, id: string, props: LayerVersionRefProps): LayerVersionRef {
+    return new ImportedLayerVersionRef(parent, id, props);
+  }
+
+  /**
+   * The ARN of the Lambda Layer version that this Layer defines.
+   */
+  public abstract readonly layerVersionArn: string;
+
+  /**
+   * The runtimes compatible with this Layer.
+   */
+  public abstract readonly compatibleRuntimes?: Runtime[];
+
+  /**
+   * Grants usage of this layer to specific entities. Usage within the same account where the layer is defined is always
+   * allowed and does not require calling this method. Note that the principal that creates the Lambda function using
+   * the layer (for example, a CloudFormation changeset execution role) also needs to have the
+   * ``lambda:GetLayerVersion`` permission on the layer version.
+   *
+   * @param awsAccountId   the AWS account ID that will be granted access to the layer, or '*' to grant usage to any/all
+   *                       AWS accounts. The default is '*'.
+   * @param organizationId when using '*' as the ``awsAccountId``, a organization ID can be procided to restrict usage
+   *                       to accounts that are memeber of a given organization.
+   */
+  public grantUsage(awsAccountId: string = '*', organizationId?: string): this {
+    if (organizationId != null && awsAccountId !== '*') {
+      throw new Error(`OrganizationId can only be specified if AwsAccountId is '*', but it is ${awsAccountId}`);
+    }
+
+    new cdk.Resource(this, `grant-usage-${awsAccountId}-${organizationId || '*'}`, {
+      type: 'AWS::Lambda::LayerVersionPermission',
+      properties: {
+        Action: 'lambda:GetLayerVersion',
+        LayerVersionArn: this.layerVersionArn,
+        Principal: awsAccountId,
+        OrganizationId: organizationId,
+      }
+    });
+    return this;
+  }
+
+  /**
+   * Exports this layer for use in another Stack. The resulting object can be passed to the ``LayerRef.import`` function
+   * to obtain a ``LayerRef`` in the user stack.
+   */
+  public export(): LayerVersionRefProps {
+    return {
+      layerVersionArn: new cdk.Output(this, 'LayerVersionArn', { value: this.layerVersionArn }).makeImportValue().toString(),
+      compatibleRuntimes: this.compatibleRuntimes,
```

Comment:
That's interesting? What's the rational of returning "compatibleRuntimes" without an Output and in general?

Reply:
The `compatibleRuntimes` property is not exposed by the resource - it's passed in as a property only. I want to be able to capture it on usage sites as well (to verify you don't build an incompatible `lambda.Function` usage). I don't want to use an output because this is a value that is available to me not as a token, and I lose a lot of value if I allow it to be tokenized (aka. - I cannot validate anymore).

Reply:
So it would only apply when you are importing/exporting within the same app, but if you export and use this from a different app, this information is not going to be used, correct?

Reply:
You can still (manually) pass the values in import and they'll still be used... But you can also omit them in this case, and no validation will be performed at synthesis time.

Reply:
ok. makes sense. interesting idea. It like the concept of more fidelity for app-local imports.